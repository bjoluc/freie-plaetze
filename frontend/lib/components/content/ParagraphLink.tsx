import React from "react";

import {Link} from "../next/Link";

interface ParagraphLinkProps {
	paragraph: string;
	children?: React.ReactNode;
}

export const ParagraphLink: React.FC<ParagraphLinkProps> = ({paragraph, children}) => (
	<Link
		href={`https://www.gesetze-im-internet.de/sgb_8/__${paragraph}.html`}
		textDecoration="underline"
		textDecorationStyle="dotted"
		target="_blank"
	>
		{children ?? <>ยง&nbsp;{paragraph} SGB&nbsp;VIII</>}
	</Link>
);
