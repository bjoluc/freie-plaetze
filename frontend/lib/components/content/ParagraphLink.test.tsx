import React from "react";
import {render} from "test-utils";

import {ParagraphLink} from "./ParagraphLink";

it("renders a link to a paragraph URL", () => {
	const {container} = render(<ParagraphLink paragraph="1">Content</ParagraphLink>);
	expect(container).toHaveTextContent("Content");
	expect(container).toMatchSnapshot();
});

it("falls back to a default link text when no children are provided", () => {
	const {container} = render(<ParagraphLink paragraph="1" />);
	expect(container).toHaveTextContent("ยง 1 SGB VIII");
	expect(container).toMatchSnapshot();
});
