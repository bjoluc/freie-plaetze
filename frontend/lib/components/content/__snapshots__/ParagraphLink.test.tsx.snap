// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`falls back to a default link text when no children are provided 1`] = `
<div>
  <a
    class="chakra-link css-1yoq2sr"
    href="https://www.gesetze-im-internet.de/sgb_8/__1.html"
    target="_blank"
  >
    § 
    1
     SGB VIII
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`renders a link to a paragraph URL 1`] = `
<div>
  <a
    class="chakra-link css-1yoq2sr"
    href="https://www.gesetze-im-internet.de/sgb_8/__1.html"
    target="_blank"
  >
    Content
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
